lib/HttpSignatureStrategy.js

```diff
@@ -171,3 +176,77 @@ function _timestampBeforeNow(timestamp) {
   var tsDate = new Date(timestamp);
   return tsDate < now;
 }
+
+function getPublicKey(publicKey, callback) {
+  var dereferenceUrlScheme = bedrock.config.passport.strategies.signature
+    .dereferenceUrlScheme;
+
+  async.waterfall([
```

Comment:
I think it's a little strange to go through this whole waterfall trying each URL scheme. It would be better to just test for the scheme here in this function in a simple if/else if/switch statement and then call the appropriate function (no waterfall needed at all).